internal/controllers/clustercatalog_controller.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+)
+
+type CatalogCacheRemover interface {
+	Remove(catalogName string) error
+}
+
+// ClusterCatalogReconciler reconciles a ClusterCatalog object
+type ClusterCatalogReconciler struct {
+	client.Client
+	Cache CatalogCacheRemover
+}
+
+//+kubebuilder:rbac:groups=olm.operatorframework.io,resources=clustercatalogs,verbs=get;list;watch
+
+func (r *ClusterCatalogReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	existingCatalog := &catalogd.ClusterCatalog{}
+	err := r.Client.Get(ctx, req.NamespacedName, existingCatalog)
+	if client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, err
+	}
+	if apierrors.IsNotFound(err) {
+		return ctrl.Result{}, r.Cache.Remove(req.Name)
+	}
+
+	return ctrl.Result{}, nil
```

Comment:
Nit: it took me an extra second to parse the `IgnoreNotFound` + `IsNotFound` checks. We could simplify to the following.
```suggestion
	if apierrors.IsNotFound(err) {
		return ctrl.Result{}, r.Cache.Remove(req.Name)
	}
	return ctrl.Result{}, err
```

If you aren't a fan of `return ctrl.Result{}, err` at the end serving the dual purpose of returning nil or non-nil errors, we could also refactor like this:
```suggestion
	if apierrors.IsNotFound(err) {
		return ctrl.Result{}, r.Cache.Remove(req.Name)
	}
	if err != nil {
		return ctrl.Result{}, err
	}
	return ctrl.Result{}, nil
```

Reply:
Do we need the `Get()` call? In our `SetupWithManager()` method below we filter events to only care about the delete events, so reconcile should only ever be called when a delete event is triggered for a `ClusterCatalog` resource.

Reply:
Yeah, true. We could make it that simple for now.

I'm guessing we will follow-up at some point to do what I mentioned here: https://github.com/operator-framework/operator-controller/pull/1207#issuecomment-2326975817

But the code is so simple as is that it would be easy to refactor later without accidentally introducing bugs or causing a regression of this code.

Reply:
I went with the second suggestion for error handling for the same reason: will be adding more into this controller soon.